compiler/rustc_middle/src/traits/mod.rs

```diff
@@ -256,20 +249,20 @@ pub enum ObligationCauseCode<'tcx> {
 
     /// Must satisfy all of the where-clause predicates of the
     /// given item.
-    ItemObligation(DefId),
+    WhereClause(DefId),
 
-    /// Like `ItemObligation`, but carries the span of the
+    /// Like `WhereClause`, but carries the span of the
     /// predicate when it can be identified.
-    BindingObligation(DefId, Span),
+    SpannedWhereClause(DefId, Span),
```

Comment:
could you instead change this to `WhereClause(DefId, Option<Span>)`? and change the matches to `WhereClause(whatever, Some(span))`?